### PR TITLE
Do not offer to use ?. when in an query that will become an expression tree

### LINKED ISSUE #17623: IDE0031: Null check can be simplified used on Expression is not valid
**Version Used**: 
Visual Studio 2017 RTM (Visual Studio/15-RTW+26228.4)
**Steps to Reproduce**:

1. Have an Expression Lambda method with a nullable check in it
2. VS 2017 suggests using IDE0031 (Null check can be simplified)
3. Auto apply fix as suggested by VS 2017 (Null type propagation)

**Expected Behavior**:
IDE0031 is not suggested
**Actual Behavior**:
IDE0031 is suggested and results in broken code after application:
-> CS8072	An expression tree lambda may not contain a null propagating operator.

Repro source:
```csharp
using System;
using System.Collections.Generic;
using System.Linq;
using System.Linq.Expressions;
using System.Text;
using System.Threading.Tasks;

namespace IDE0031Repro
{
    class Program
    {
        static void Main(string[] args)
        {
            var test = new Test<DTO>();
            test.Method(t => t.NullableGuid != null ? t.NullableGuid.ToString() : null); // works
            test.Method(t => t.NullableGuid?.ToString()); // broken, transformed from above with IDE0031
        }
    }

    class DTO
    {
        public Guid? NullableGuid { get; set; }
    }

    class Test<T>
    {
        public void Method(Expression<Func<T, string>> functor)
        {
        }
    }
}

```

### LINKED ISSUE #33992: 'IDE0031 Use null propagation' incorrectly suggested for LINQ expression
**Version Used**: VS 2019 RC.1 SVC1

Seems to be a repeat of https://github.com/dotnet/roslyn/issues/17623 but when using LINQ syntax.

```cs
_ = from item in new[] { new { X = (int?)1 } }.AsQueryable()
    //     ↓ IDE0031 Use null propagation
    select item == null ? null : item.X;

_ = from item in Enumerable.Empty<(int? x, int? y)?>().AsQueryable()
    //     ↓ IDE0031 Use null propagation
    select item == null ? null : item.Value.x;
```

After applying suggested fix:

```cs
_ = from item in new[] { new { X = (int?)1 } }.AsQueryable()
    //     ↓ CS8072 An expression tree lambda may not contain a null propagating operator.
    select item?.X;

_ = from item in Enumerable.Empty<(int? x, int? y)?>().AsQueryable()
    //     ↓ CS8072 An expression tree lambda may not contain a null propagating operator.
    select item?.x;
```

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes  #33992
Fixes #17623

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs [hunk 1]
```diff
@@ -906,7 +907,7 @@ public static bool ContainsInArgument(this ConstructorInitializerSyntax initiali
                 return false;
             }
 
-            return initializer.ArgumentList.Arguments.Any(a => a.Span.Contains(textSpan));
+            return initializer.ArgumentList.Arguments.Any<ArgumentSyntax>(a => a.Span.Contains(textSpan));
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs]:
```suggestion
            return initializer.ArgumentList.Arguments.Any(a => a.Span.Contains(textSpan));
```

**Comment by dibarbet** [comment]
@CyrusNajmabadi did you mean to link #33992 and #17623 ?  #26342 doesn't look like it applies

**Comment by CyrusNajmabadi** [comment]
> @CyrusNajmabadi did you mean to link

Yup.  Thanks.  Want to review? :)

**Comment by dibarbet** [comment]
> Yup. Thanks. Want to review? :)

I'll be honest I need to study this one a bit more to figure out the bug and why this is the correct change - my linq understanding could use some work.  I will review, but I would definitely suggest another pair of eyes.

**Comment by ryzngard** [comment]
It's worth having another set of eyes on this. The behavior looks correct, the code is easy enough to follow, but I'm less familiar with this area.

**Comment by CyrusNajmabadi** [comment]
> It's worth having another set of eyes on this. The behavior looks correct, the code is easy enough to follow, but I'm less familiar with this area.

it's pretty simple in concept.  Query expressions are translated to calls to methods (literally through a syntactic transformation as per hte spec).  So we use the compiler helper methods to ask semantic questions about those method calls.  I just looked at the compiler API (and impl) to enumerate the query operations it exposed to see if the data was being converted to an expression tree for those symbols.

**Comment by CyrusNajmabadi** [comment]
thanks!

**Pull request merged**